common/src/Model/Entity.cpp

```diff
@@ -150,30 +150,31 @@ namespace TrenchBroom {
         }
 
         void Entity::doChildWasAdded(Node* node) {
-            nodeBoundsDidChange();
+            nodeBoundsDidChange(bounds());
         }
         
         void Entity::doChildWasRemoved(Node* node) {
-            nodeBoundsDidChange();
+            nodeBoundsDidChange(bounds());
         }
 
-        void Entity::doNodeBoundsDidChange() {
+        void Entity::doNodeBoundsDidChange(const BBox3& oldBounds) {
             invalidateBounds();
         }
         
-        void Entity::doChildBoundsDidChange(Node* node) {
-            nodeBoundsDidChange();
+        void Entity::doChildBoundsDidChange(Node* node, const BBox3& oldBounds) {
+            const BBox3& myBounds = bounds();
+            if (!myBounds.encloses(oldBounds) && !myBounds.encloses(node->bounds())) {
```

Comment:
What if the oldBounds was contributing to this's bounds, but the new child bounds `node->bounds()` aren't? Is it correct to not call nodeBoundsDidChange?

Reply:
I think it's correct now. We recompute the bounds of the entity (and now also the group) and compare the new bounds to the old bounds. Only if the new bounds are different from the old bounds do we call nodeBoundsDidChange().